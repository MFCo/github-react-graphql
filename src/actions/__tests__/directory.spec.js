import configureStore from 'redux-mock-store';
import act from '../../actions';

const middlewares = [];
const mockStore = configureStore(middlewares);

it('should dispatch action to update directory', () => {

  const initialState = {};
  const store = mockStore(initialState);

  store.dispatch(act.updateDirectory({
    newList:
      [
        {
          "name": ".babelrc",
          "type": "blob",
          "oid": "d42ea8be8b19750c64dc2a305cbd198739e9adbe"
        },
        {
          "name": ".dockerignore",
          "type": "blob",
          "oid": "c8f50f7cd317669d25aedba7415f8a1cbaadbab0"
        },
        {
          "name": ".eslintrc.js",
          "type": "blob",
          "oid": "29e7593c8ba8c3aedfcf2f78750e43ec428e367f"
        }
      ],
    newId: 'c8f50f7cd317669d25aedba7415f8a1cbaadbab0'
  }));

  const actions = store.getActions();
  expect(actions).toMatchSnapshot();
});

it('should dispatch action to forward on stack trace', () => {

  const initialState = {};
  const store = mockStore(initialState);

  store.dispatch(act.forwardStack({
    newStack: ['c8f50f7cd317669d25aedba7415f8a1cbaadbab0']
  }));

  const actions = store.getActions();
  expect(actions).toMatchSnapshot();
});

it('should dispatch action to backward on stack trace', () => {

  const initialState = {};
  const store = mockStore(initialState);

  store.dispatch(act.backwardStack());

  const actions = store.getActions();
  expect(actions).toMatchSnapshot();
});
