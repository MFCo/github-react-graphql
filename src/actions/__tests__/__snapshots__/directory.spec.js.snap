// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should dispatch action to backward on stack trace 1`] = `
Array [
  Object {
    "type": "BACKWARD_STACK",
  },
]
`;

exports[`should dispatch action to forward on stack trace 1`] = `
Array [
  Object {
    "payload": Object {
      "newStack": Array [
        "c8f50f7cd317669d25aedba7415f8a1cbaadbab0",
      ],
    },
    "type": "FORWARD_STACK",
  },
]
`;

exports[`should dispatch action to update directory 1`] = `
Array [
  Object {
    "payload": Object {
      "newId": "c8f50f7cd317669d25aedba7415f8a1cbaadbab0",
      "newList": Array [
        Object {
          "name": ".babelrc",
          "oid": "d42ea8be8b19750c64dc2a305cbd198739e9adbe",
          "type": "blob",
        },
        Object {
          "name": ".dockerignore",
          "oid": "c8f50f7cd317669d25aedba7415f8a1cbaadbab0",
          "type": "blob",
        },
        Object {
          "name": ".eslintrc.js",
          "oid": "29e7593c8ba8c3aedfcf2f78750e43ec428e367f",
          "type": "blob",
        },
      ],
    },
    "type": "UPDATE_DIRECTORY",
  },
]
`;
